run.py

```diff
@@ -24,35 +26,36 @@ def build_image():
     `/input/collection/collection_name`, where the container can
     expect the document collection to be mounted.
     """
-    base = client.containers.run("{}:{}".format(args.repo, args.tag), command="sh -c '/init; /index {}'".format(args.collection_name), volumes=volumes, detach=True)
-    base.wait()
-    base.commit(repository=args.repo, tag="save")
 
+    print("Preparing image...")
 
-if __name__ == "__main__":
+    volumes = {
+        args.collection_path: {
+            "bind": os.path.join(COLLECTION_PATH_GUEST, args.collection_name),
+            "mode": "ro"
+        }
+    }
 
-    parser = argparse.ArgumentParser()
-    parser.add_argument("--repo", required=True, type=str, help="the image repo (i.e., anserini)")
-    parser.add_argument("--tag", required=True, type=str, help="the image tag (i.e., latest)")
-    parser.add_argument("--collection_name", required=True, type=str, help="the name of the collection")
-    parser.add_argument("--collection_path", required=True, type=str, help="the path of the collection on the host")
-    parser.add_argument("--output", required=True, type=str, help="the output directory for run files on the host")
-    parser.add_argument("--topic", required=True, type=str, help="the topic file for search")
-    parser.add_argument("--topic_format", default="TREC", type=str, help="the topic file for search")
-    parser.add_argument("--qrels", required=True, type=str, help="the qrels file for evaluation")
-    parser.add_argument("--build", default=False, type=bool, help="whether we re-build the image from scratch")
+    # The first step is to pull an image from an OSIRRC participant,
+    # start up a container, run its `init` and `index` hooks, and then
+    # use `docker commit` to save the image after the index has been
+    # built. The rationale for doing this is that indexing may take a
+    # while, but only needs to be done once, so in essence we are
+    # "snapshotting" the system with the indexes.
+    base = client.containers.run("{}:{}".format(args.repo, args.tag), command="sh -c '/init; /index {}'".format(args.collection_name), volumes=volumes, detach=True)
+    base.wait()
+    base.commit(repository=args.repo, tag="save")
 
-    # Parse the args
-    args = parser.parse_args()
 
-    # Create Docker client
-    client = docker.from_env()
+def search():
+    """
+    Runs the search and evaluates the results (run files placed into the /output directory) using trec_eval
+    """
+    exists = len(client.images.list(filters={"reference": "{}:{}".format(args.repo, "save")}))
+    if not exists:
```

Comment:
For sake of clarify, I would write something like `exists = len(...) != 0`, but it's just personal preference.